compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1327,6 +1327,11 @@ pub struct TyCtxt<'tcx> {
     gcx: &'tcx GlobalCtxt<'tcx>,
 }
 
+// Explicitly implement `DynSync` and `DynSend` for `TyCtxt` to short circuit trait resolution. Its
+// field are asserted to implement these traits below, so this is trivially safe, and it greatly
+// speeds-up compilation of this crate.
```

Comment:
Thanks!

I think this not just speeds up this crate's build. Because there are other places that use parallel interfaces (such as typeck) need to check `TyCtxt: DynSend/DynSync`.

Reply:
Yea, that's OK

Reply:
Done.